### PR TITLE
Refactorings to WS + N trackers + remove global BOOTNODES #132

### PR DESCRIPTION
**Pull request by harbu**
Did the refactorings we talked about in PR and meeting.

### CODE REVIEW

File: src/connection/WsEndpoint.js [hunk 1]
```diff
@@ -3,15 +3,13 @@ const { EventEmitter } = require('events')
...
 const debug = require('debug')('streamr:connection:ws-endpoint')
 const WebSocket = require('ws')
-const uuidv4 = require('uuid/v4')
 
 const Endpoint = require('./Endpoint')
 
-module.exports = class WsEndpoint extends EventEmitter {
-    constructor(node, id) {
+class WsEndpoint extends EventEmitter {
+    constructor(node) {
```
Comment (mirpo) [inline src/connection/WsEndpoint.js]: node -> wss

File: src/connection/WsEndpoint.js [hunk 2]
```diff
@@ -20,87 +18,93 @@ module.exports = class WsEndpoint extends EventEmitter {
 
         this.wss.on('connection', (ws, req) => {
             const parameters = url.parse(req.url, true)
-            const peerId = parameters.query.address
-
-            this._onConnected(ws, peerId)
+            const { address } = parameters.query
```
Comment (mirpo) [inline src/connection/WsEndpoint.js:21]: nice

File: src/connection/WsEndpoint.js [hunk 3]
```diff
@@ -20,87 +18,93 @@ module.exports = class WsEndpoint extends EventEmitter {
...
+    send(recipientAddress, message) {
+        if (!this.isConnected(recipientAddress)) {
+            debug('cannot send to %s because not in peer book', recipientAddress)
             return false
         }
 
+        const ws = this.connections.get(recipientAddress)
+
         try {
             ws.send(message)
```
Comment (mirpo) [inline src/connection/WsEndpoint.js]: later need more check async errors (with callback)
like ws.send('something',(err) => { if (err) ....

File: src/connection/WsEndpoint.js [hunk 4]
```diff
@@ -20,87 +18,93 @@ module.exports = class WsEndpoint extends EventEmitter {
...
-                this._onConnected(ws, peerAddress)
-            })
-            ws.on('error', (err) => {
-                debug('failed to connect to %s, error: %o', peerAddress, err)
-            })
-        } catch (err) {
-            debug('failed to connect to %s, error: %o', peerAddress, err)
-        }
+        return new Promise((resolve, reject) => {
+            if (this.connections.has(peerAddress)) {
```
Comment (mirpo) [inline src/connection/WsEndpoint.js]: why not isConnected? like in send?

**Review by mirpo — APPROVED** [review]
fill free to merge, only minor stuff

File: src/connection/WsEndpoint.js (hunk 3 above)
Comment (harbu) [reply src/connection/WsEndpoint.js]: fixed

File: src/connection/WsEndpoint.js (hunk 4 above)
Comment (harbu) [reply src/connection/WsEndpoint.js]: fixed

File: src/connection/WsEndpoint.js (hunk 1 above)
Comment (harbu) [reply src/connection/WsEndpoint.js]: fixed

**Review by mirpo — APPROVED** [review]

**Pull request merged**